app/components/teachers/record_outcome_component.rb

```diff
@@ -63,5 +55,9 @@ def unsupported_service?
       !service.is_a?(::AppropriateBodies::RecordPass) &&
         !service.is_a?(::AppropriateBodies::RecordFail)
     end
+
+    def failed?
```

Comment:
The original predicate might be more resilient, if you only need confirmation in some modes we can combine them:

```ruby
def confirm_fail? = failed? && appropriate_body_mode?
```